test/unit/lib/population_clone_job_test.rb

```diff
@@ -14,6 +14,105 @@ def test_perform_full_deck
     assert_equal 10, Patient.where(correlation_id: @pt.id).count
   end
 
+  def test_preferred_code_procedure_without_vendor_perference
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    original_patient = @pt.patients.first
+    original_procedure_codes = original_patient.qdmPatient.procedures.first.dataElementCodes
+    # There are two codes in the original patients procedure
+    assert_equal 2, original_procedure_codes.size
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+  end
+
+  def test_preferred_code_procedure_with_vendor_perference_snomedct
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = ['SNOMEDCT']
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should be from the SNOMEDCT code system
+    assert_equal 'SNOMEDCT', cloned_procedure_codes[0].codeSystem
+  end
+
+  def test_preferred_code_procedure_with_vendor_perference_cpt
+    pcj = Cypress::PopulationCloneJob.new('test_id' => @pt.id)
+    pcj.find_patients_to_clone
+    vendor = @pt.product.vendor
+    vendor.preferred_code_systems['procedure'] = ['CPT']
+    vendor.save
+    @pt.patients
+    original_patient = @pt.patients.first
+    cloned_patient = original_patient.clone
+    pcj.restrict_entry_codes(cloned_patient)
+    cloned_procedure_codes = cloned_patient.qdmPatient.procedures.first.dataElementCodes
+    # There should only be one codes in the cloned patients procedure
+    assert_equal 1, cloned_procedure_codes.size
+    # The cloned code should be from the CPT code system
+    assert_equal 'CPT', cloned_procedure_codes[0].codeSystem
+  end
+
```

Comment:
The snomedct and cpt functions are nearly identical except for passing in and checking the code. It might be worth thinking about splitting most of the code into a different function for DRY.